locations/commands/genspider.py

```diff
@@ -0,0 +1,146 @@
+                    elif "operator:wikidata" in nsi_matches[0]["tags"].keys():
+                        self.parameters["operator_wikidata"] = wikidata_code
+
+    # TODO: Remove this when autospidergen is merged
+    def automatically_set_parameters(self):
+        """
+        Automatically extract parameters from at least one or more
+        of the following:
+          1. start_urls[0]
+          2. parameters["brand_wikidata"]
+          3. parameters["operator_wikidata"]
+
+        If any of the above are specified for this spider, and this
+        method is called, this method will attempt to populate all
+        other parameters automatically from Name Suggestion Index
+        data.
+
+        See automatically_set_brand_or_operator_from_start_url(self)
+        for details on how parameters are extracted from just a single
+        supplied start_urls[0].
+        """
+        if not self.parameters["brand_wikidata"] and not self.parameters["operator_wikidata"]:
+            self.automatically_set_brand_or_operator_from_start_url()
+
+        nsi = NSI()
+        if self.parameters["brand_wikidata"] or self.parameters["operator_wikidata"]:
+            wikidata_code = self.parameters["brand_wikidata"] or self.parameters["operator_wikidata"]
+            nsi_matches = [nsi_match for nsi_match in nsi.iter_nsi(wikidata_code)]
+            if len(nsi_matches) != 1:
+                return
+
+            if found_keys := NSI.generate_keys_from_nsi_attributes(nsi_matches[0]):
+                self.parameters["spider_key"] = found_keys[0]
+                self.parameters["spider_class_name"] = found_keys[1]
+
+            if self.parameters["brand_wikidata"]:
+                brand = nsi_matches[0]["tags"].get("brand", nsi_matches[0]["tags"].get("name"))
+                self.parameters["brand"] = brand
+            elif self.parameters["operator_wikidata"]:
+                operator = nsi_matches[0]["tags"].get("operator", nsi_matches[0]["tags"].get("name"))
+                self.parameters["operator"] = operator
+
+    def _generate_template_variables(
+        self,
+        module: str,
+        name: str,
+        url: str,
+        template_name: str,
+    ):
+        capitalized_module = "".join(s.capitalize() for s in module.split("_"))
+
+        self.start_urls = [url]
+        self.automatically_set_parameters()
+
+        tvars = {
+            "project_name": self.settings.get("BOT_NAME"),
+            "ProjectName": string_camelcase(self.settings.get("BOT_NAME")),
+            "module": module,
+            "name": name,
+            "url": url,
+            "domain": extract_domain(url),
+            "classname": f"{capitalized_module}Spider",
+        }
+        # Flatten the parameters and merge with the other variables
+        return tvars | self.parameters
+
+    # TODO: If https://github.com/scrapy/scrapy/pull/6470 is merged, much of this can be removed
```

Comment:
This did get merged, need to wait for release or point it git version and remove this method